src/tools/compiletest/src/runtest.rs

```diff
@@ -1905,13 +1905,14 @@ impl<'test> TestCx<'test> {
         match output_file {
             TargetLocation::ThisFile(path) => {
                 rustc.arg("-o").arg(path);
+                rustc.arg("--out-dir").arg(&self.config.build_base);
             }
             TargetLocation::ThisDirectory(path) => {
                 if is_rustdoc {
                     // `rustdoc` uses `-o` for the output directory.
                     rustc.arg("-o").arg(path);
                 } else {
-                    rustc.arg("--out-dir").arg(path);
+                    rustc.arg("--out-dir").arg(&self.config.build_base);
```

Comment:
These are not the same. `build_base` is the same for all tests in the suite, while `path` is based on the test name:
```
[src/tools/compiletest/src/runtest.rs:1914] &path = "/home/jnelson/rust-lang/rust/build/x86_64-unknown-linux-gnu/test/ui/const-generics/const-argument-non-static-lifetime.min"
[src/tools/compiletest/src/runtest.rs:1914] &self.config.build_base = "/home/jnelson/rust-lang/rust/build/x86_64-unknown-linux-gnu/test/ui"
```
The difference matters when - like in this case - the test creates a subdirectory in the out-dir. If multiple tests create `/home/jnelson/rust-lang/rust/build/x86_64-unknown-linux-gnu/test/ui/save-analysis`, then it's impossible to test the output, because it will non-deterministic when tests are run in parallel. I think we should keep using `path` here.